feat: Track User Registration Source for Google Extended Access

AnuragVasanwala opened the pull request:
## Description

This pull request adds tracking for user registration source, specifically for users who register via Google Extended Access. 

### Key Changes:

- A new user meta field, `registered_by`, is added to users who successfully register using Google Extended Access.
- The `registered_by` field will store the value `google-extended-access` to identify these users.

Fixes: #7

AnuragVasanwala committed:
feat: Add tracking for registration source

Add user meta data to track the registration source when a user is created. This commit adds the 'registered_by' meta key with the value 'google-extended-access' to the newly created user.

leogermani reviewed (changes requested):
Hi!

There is already a place we use to store the "registration_method" when registering a user via the `Reader_Activation` class.

You can pass it - and we probably should be passing already, in the `$metadata` argument of the `register_method`

```
$result = Newspack\Reader_Activation::register_reader( $email, '', true, [ 'registration_method' => 'google-extended-access' ] );
```

Can you do that change please?

The info will be available in the `np_reader_registration_method` user meta (see `Newspack\Reader_Activation::REGISTRATION_METHOD`

Thank you!

AnuragVasanwala committed:
♻️ Refactor `registered_by` to implement `registration_method`

Instead of adding a new user meta, implemented `registration_method` to mark user as registered by `google-extended-access`.

AnuragVasanwala commented:
> Hi!
> 
> There is already a place we use to store the "registration_method" when registering a user via the `Reader_Activation` class.
> 
> You can pass it - and we probably should be passing already, in the `$metadata` argument of the `register_method`
> 
> ```
> $result = Newspack\Reader_Activation::register_reader( $email, '', true, [ 'registration_method' => 'google-extended-access' ] );
> ```
> 
> Can you do that change please?
> 
> The info will be available in the `np_reader_registration_method` user meta (see `Newspack\Reader_Activation::REGISTRATION_METHOD`
> 
> Thank you!

Hi @leogermani 👋🏻

Thank you for the suggestion ✨

I've updated the `register_reader` function to include the `registration_method` in the `$metadata` array as discussed (Commit Ref: https://github.com/Automattic/newspack-extended-access/pull/8/commits/1e6e46069626527f5edd5877f9613f65ebcb94ca). Would you please review PR whenever time permits?
